nuxeo-features/nuxeo-platform-notification/nuxeo-platform-notification-core/src/test/java/org/nuxeo/ecm/platform/ec/notification/TestEmailNotification.java

```diff
@@ -0,0 +1,171 @@
+import org.nuxeo.ecm.platform.test.PlatformFeature;
+import org.nuxeo.mail.SmtpMailServerFeature;
+import org.nuxeo.runtime.test.runner.Deploy;
+import org.nuxeo.runtime.test.runner.Features;
+import org.nuxeo.runtime.test.runner.FeaturesRunner;
+import org.nuxeo.runtime.test.runner.TransactionalFeature;
+
+/**
+ * Tests the whole email notification process: from subscribing to a given event to the reception of the email for this
+ * event.
+ *
+ * @since 11.1
+ */
+@RunWith(FeaturesRunner.class)
+@Features({ PlatformFeature.class, SmtpMailServerFeature.class })
+@Deploy("org.nuxeo.ecm.platform.notification.core")
+@Deploy("org.nuxeo.ecm.platform.notification.api")
+@Deploy("org.nuxeo.ecm.platform.url.api")
+@Deploy("org.nuxeo.ecm.platform.url.core")
+@Deploy("org.nuxeo.ecm.platform.notification.core.tests:OSGI-INF/notification-event-listener-contrib.xml")
+public class TestEmailNotification {
+
+    protected static final String DUMMY_NOTIFICATION_NAME = "DummyNotificationToSendMail";
+
+    protected static final String DUMMY_EVENT_NAME = "dummyNotificationToSendMail";
+
+    protected static final String DOCUMENT_NAME = "anyFile";
+
+    protected static final SimpleDateFormat EVENT_DATE_FORMAT = new SimpleDateFormat("dd/MM/yyyy - HH:mm");
+
+    @Inject
+    protected CoreSession session;
+
+    @Inject
+    protected NotificationManager notificationManager;
+
+    @Inject
+    protected EventService eventService;
+
+    @Inject
+    protected TransactionalFeature transactionalFeature;
+
+    @Inject
+    protected SmtpMailServerFeature.MailsResult emailsResult;
+
+    protected DocumentModel domain;
+
+    @Before
+    public void before() {
+        domain = session.createDocumentModel("/", "domain", "Domain");
+        domain = session.createDocument(domain);
+    }
+
+    @Test
+    public void shouldReceiveNotificationMailWhenSubscribeToDocument() {
+        DocumentModel documentModel = session.createDocumentModel(domain.getPathAsString(), DOCUMENT_NAME, "File");
+        documentModel = session.createDocument(documentModel);
+
+        addDummySubscription(documentModel);
+
+        DocumentEventContext ctx = new DocumentEventContext(session, session.getPrincipal(), documentModel);
+        Event event = ctx.newEvent(DUMMY_EVENT_NAME);
+        eventService.fireEvent(event);
+        transactionalFeature.nextTransaction();
+
+        checkMailContent(documentModel, event);
+    }
+
+    @Test
+    public void shouldReceiveNotificationMailWhenSubscribeToParentDocument() {
+        DocumentModel folder = session.createDocumentModel(domain.getPathAsString(), "anyFolder", "Folder");
+        folder = session.createDocument(folder);
+
+        addDummySubscription(folder);
+
+        // create a simple hierarchy (domain <- anyFolder <- DOCUMENT_NAME)
+        DocumentModel mainDocModel = session.createDocumentModel(folder.getPathAsString(), DOCUMENT_NAME, "File");
+        mainDocModel = session.createDocument(mainDocModel);
```

Comment:
Could you remove the comment and replace the createDocumentModel line by a real path which will show us the hierarchy:
```java
        DocumentModel mainDocModel = session.createDocumentModel("/domain/anyFolder", DOCUMENT_NAME, "File");
        mainDocModel = session.createDocument(mainDocModel);
```